### PR TITLE
xtest: inline description of test dependencies

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by etienne-lms** [comment]
1st commit is related to discussion thread https://github.com/OP-TEE/optee_test/pull/427#issuecomment-630180482.
The 2nd commit is an kind of extension for all other tests that can be skipped based on unsatisfied dependencies from TEE resources.

**Comment by b49020** [comment]
`Acked-by: Sumit Garg <sumit.garg@linaro.org>`

**Comment by etienne-lms** [comment]
Tag applied.

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -5549,6 +5549,11 @@ static void xtest_tee_test_4013(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(regression, 4013, xtest_tee_test_4013,
 		"Test generation of device unique TA keys");
 
+/*
+ * regression_4014
+ * Depends on SM2 KEP support in OP-TEE OS.
```
Comment (jforissier) [inline host/xtest/regression_4000.c:5554]: True, and sub-cases in 4006 depend on SM2 PKE or SM2 DSA yet you're not mentioning that... Similarly 4001 will test SM3 if available.

In fact I don't see a reason to add all these comments except for the kernel-related ones to help identify which patches exactly are needed.

Comment (etienne-lms) [reply host/xtest/regression_4000.c:5554]: I agree :) that's why I split into 2 commits.
I propose to discard the 2nd commit and keep only the 1st one. Ok?

Comment (jforissier) [reply host/xtest/regression_4000.c:5554]: > I agree :) that's why I split into 2 commits.

Aha true I missed that :) it's good.

> I propose to discard the 2nd commit and keep only the 1st one. Ok?

Sounds good.

**Comment by jenswi-linaro** [comment]
Same here for "xtest: regression 1027, 1028: add a comment on test dependencies ":
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Rephrased commit message: 
```patch
-Add an inline comment on dependencies on regression tests 1027 and 1028.
-When these dependencies are not satisfied, xtest fails.
+Add an inline comments on kernel patch dependencies on regression tests
+1027 and 1028. When these dependencies are not satisfied, xtest fails.

```
Tags applied.

**Pull request merged**